### PR TITLE
Make nvcc use conda gcc when possible

### PR DESCRIPTION
**Pull request by leshikus**
Subj

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -24,8 +24,11 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
-        with:
-          submodules: true
+
+      - name: Free space
+        run: |
+          sudo rm -rf /usr/local/lib/android/sdk /opt/hostedtoolcache /usr/share/swift /opt/az
+          sudo apt-get purge -y gcc
```
Comment (Garra1980) [inline .github/workflows/build.yml:31]: non-related to Free space step?

Comment (leshikus) [reply .github/workflows/build.yml:31]: It's related, I remove gcc for conda build. As for the "rm" line, I will try to keep this step the same for all containers (and this removal is needed for docker containers).

Comment (leshikus) [reply .github/workflows/build.yml:31]: Now if someone again will add the call to OS gcc (not conda gcc), then the build will report that

Comment (leshikus) [reply .github/workflows/build.yml:31]: Should I also include this into PR https://github.com/intel-ai/hdk/commit/25adfaa10f466c3a94b1412d4de92f1e983a86f1 ?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -70,12 +73,13 @@ jobs:
           CPU_COUNT: 2
           RUN_TESTS: 1
         run: |
+          sh -vx
```
Comment (Garra1980) [inline .github/workflows/build.yml:76]: debugging stuff?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -62,3 +62,4 @@ jobs:
 
   sanity-test:
     uses: ./.github/workflows/build-docker-hdk.yml
+
```
Comment (Garra1980) [inline .github/workflows/main.yml:65]: let's remove this

**Pull request merged**